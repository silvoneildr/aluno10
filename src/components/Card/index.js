import React from 'react';
import propTypes from 'prop-types';

import Box from './styles';

const Card = ({ children }) => <Box>{children}</Box>;

Card.propTypes = {
  children: propTypes.node.isRequired
};

export default Card;
