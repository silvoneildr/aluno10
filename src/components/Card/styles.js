import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  height: 100%;

  background: ${props => props.theme.light};
  padding: 20px;
  border-radius: 10px;
`;

export default Container;
