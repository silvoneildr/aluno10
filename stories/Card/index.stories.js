import React from 'react';

import { storiesOf } from '@storybook/react';

import Card from '../../src/components/Card';

storiesOf('Components|Card', module).add('with text', () => <Card>Hello Button</Card>);

storiesOf('Components|Card', module).add('with p', () => (
  <Card>
    <p>Hello world</p>
  </Card>
));
