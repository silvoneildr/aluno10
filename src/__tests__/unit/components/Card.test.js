import React from 'react';
import Card from 'Components/Card';
import { render } from 'react-testing-library';

describe('<Card />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Card>
        <p>Hello World</p>
      </Card>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
