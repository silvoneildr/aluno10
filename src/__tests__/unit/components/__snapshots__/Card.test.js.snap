// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 20px;
  border-radius: 10px;
}

<div
  class="c0"
>
  <p>
    Hello World
  </p>
</div>
`;
